ex7/ex7.js

```diff
@@ -9,3 +10,59 @@
  * mostFrequentChar("a b c d e f") === " "
  * mostFrequentChar("") === null
  */
+
+function mostFrequentChar(str) {
+
+	// If str empty tmpChar will return NULL
+	var  winnerChar= null, index = 0,
+		topRepeat = { charValue: '', charCount: 0}, 
+		search = { charValue: '', charCount: 0}, 
+		doNotSearchList = [];
+
+	if (str != '') {
+
+		// Loop through all chars in str
+		for (var i = 0; i < str.length; i++) {
+
+			// The char we are now working on
+			search.charValue = str.charAt(i); 
+			search.charCount = 1; 
+
+			// We will count only the char
+			if (doNotSearchList.indexOf(search.charValue, 0) === -1) {
+
+				// Add search char to doNotSearchList 
+				// So we do not look for it next time
+				doNotSearchList.push(search.charValue);
+
+				index = 1;
+				index = str.indexOf(search.charValue, i + index);
```

Comment:
Same as in the other question -- this will be O(n*n) = O(n^2) order of n squared. Switch to object-based searching instead.